src/test/java/com/app/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,199 @@
+
+        CREATE_BOOK_REQUEST_DTO.setTitle("Test Title1");
+        CREATE_BOOK_REQUEST_DTO.setAuthor("Test Author1");
+        CREATE_BOOK_REQUEST_DTO.setCategoryIds(Set.of(1L));
+        CREATE_BOOK_REQUEST_DTO.setIsbn("978-3-16-148410-0");
+        CREATE_BOOK_REQUEST_DTO.setPrice(BigDecimal.valueOf(19.99));
+        CREATE_BOOK_REQUEST_DTO.setDescription("Test Description1");
+        CREATE_BOOK_REQUEST_DTO.setCoverImage("Test Cover Image1");
+
+        BOOK_DTO.setId(1L);
+        BOOK_DTO.setTitle(CREATE_BOOK_REQUEST_DTO.getTitle());
+        BOOK_DTO.setAuthor(CREATE_BOOK_REQUEST_DTO.getAuthor());
+        BOOK_DTO.setCategoryIds(CREATE_BOOK_REQUEST_DTO.getCategoryIds());
+        BOOK_DTO.setIsbn(CREATE_BOOK_REQUEST_DTO.getIsbn());
+        BOOK_DTO.setPrice(CREATE_BOOK_REQUEST_DTO.getPrice());
+        BOOK_DTO.setDescription(CREATE_BOOK_REQUEST_DTO.getDescription());
+        BOOK_DTO.setCoverImage(CREATE_BOOK_REQUEST_DTO.getCoverImage());
+    }
+
+    @Test
+    @WithMockUser(username = "ADMIN", roles = {"ADMIN"})
+    @DisplayName("Create a new book")
+    void createBook_validRequestDto_returnBookDto() throws Exception {
+        CreateBookRequestDto saveBookRequestDto = new CreateBookRequestDto();
+        saveBookRequestDto.setTitle(CREATE_BOOK_REQUEST_DTO.getTitle());
+        saveBookRequestDto.setAuthor(CREATE_BOOK_REQUEST_DTO.getAuthor());
+        saveBookRequestDto.setCategoryIds(CREATE_BOOK_REQUEST_DTO.getCategoryIds());
+        saveBookRequestDto.setIsbn("978-1-23-456789-7");
+        saveBookRequestDto.setPrice(CREATE_BOOK_REQUEST_DTO.getPrice());
+        saveBookRequestDto.setDescription(CREATE_BOOK_REQUEST_DTO.getDescription());
+        saveBookRequestDto.setCoverImage(CREATE_BOOK_REQUEST_DTO.getCoverImage());
+
+        BookDto bookDto = new BookDto();
+        bookDto.setTitle(saveBookRequestDto.getTitle());
+        bookDto.setAuthor(saveBookRequestDto.getAuthor());
+        bookDto.setCategoryIds(saveBookRequestDto.getCategoryIds());
+        bookDto.setIsbn(saveBookRequestDto.getIsbn());
+        bookDto.setPrice(saveBookRequestDto.getPrice());
+        bookDto.setDescription(saveBookRequestDto.getDescription());
+        bookDto.setCoverImage(saveBookRequestDto.getCoverImage());
+
+        String jsonRequest = objectMapper.writeValueAsString(saveBookRequestDto);
+
+        MvcResult result = mockMvc.perform(
+                        post("/books")
+                                .content(jsonRequest)
+                                .contentType(MediaType.APPLICATION_JSON)
+                )
+                .andExpect(status().isOk())
+                .andReturn();
+        BookDto actual = objectMapper.readValue(result
+                .getResponse().getContentAsString(), BookDto.class);
+        Assertions.assertNotNull(actual);
+        Assertions.assertNotNull(actual.getId());
+        assertTrue(EqualsBuilder.reflectionEquals(bookDto, actual, "id"));
```

Comment:
What is the reason to use 'assertTrue(EqualsBuilder.reflectionEquals(...' is it possible to assertEquals?